packages/devtools_app/lib/src/inspector/inspector_controller.dart

```diff
@@ -581,42 +580,56 @@ class InspectorController extends DisposableController
     _selectionGroups.cancelNext();
 
     final group = _selectionGroups.next;
-    final pendingSelectionFuture = group.getSelection(
-      selectedDiagnostic,
-      treeType,
-      isSummaryTree: isSummaryTree,
-    );
 
-    final Future<RemoteDiagnosticsNode> pendingDetailsFuture = isSummaryTree
-        ? group.getSelection(selectedDiagnostic, treeType, isSummaryTree: false)
-        : null;
+    if (firstFrame) {
+      final queryParams = loadQueryParams();
+      if (queryParams.containsKey(inspectorRefQueryParam)) {
+        print('trying to select from service');
+        await group.setSelectionInspector(
+          InspectorInstanceRef(queryParams[inspectorRefQueryParam]),
+          false,
+        );
```

Comment:
@jacob314 this code as is results in some errors from the flutter framework:
```
{"exception":"Id does not exist.","stack":"#0      WidgetInspectorService.toObject (package:flutter/src/widgets/widget_inspector.dart:1286:7)\n#1      WidgetInspectorService.setSelectionById (package:flutter/src/widgets/widget_inspector.dart:1348:25)\n#2      WidgetInspectorService._registerServiceExtensionWithArg.<anonymous closure> (package:flutter/src/widgets/widget_inspector.dart:867:35)\n#3      WidgetInspectorService._registerServiceExtensionWithArg.<anonymous closure> (package:flutter/src/widgets/widget_inspector.dart:864:17)\n#4      BindingBase.registerServiceExtension.<anonymous closure> (package:flutter/src/foundation/binding.dart:580:32)\n<asynchronous suspension>\n","method":"ext.flutter.inspector.setSelectionById"}
trying to select from service
Error: ext.flutter.inspector.setSelectionById: (-32000) Server error
{"exception":"Id does not exist.","stack":"#0      WidgetInspectorService.toObject (package:flutter/src/widgets/widget_inspector.dart:1286:7)\n#1      WidgetInspectorService.setSelectionById (package:flutter/src/widgets/widget_inspector.dart:1348:25)\n#2      WidgetInspectorService._registerServiceExtensionWithArg.<anonymous closure> (package:flutter/src/widgets/widget_inspector.dart:867:35)\n#3      WidgetInspectorService._registerServiceExtensionWithArg.<anonymous closure> (package:flutter/src/widgets/widget_inspector.dart:864:17)\n#4      BindingBase.registerServiceExtension.<anonymous closure> (package:flutter/src/foundation/binding.dart:580:32)\n<asynchronous suspension>\n","method":"ext.flutter.inspector.setSelectionById"}
```

Reply:
Can you print the id you are passing?

Reply:
You can also add logging to the WidgetInspectorService code to make sure that the id that got read in is what you expected.

Reply:
I am passing an id like: `instance-inspector-7`

Reply:
update - issue was with the "instance-" prefix in the id